csaf_2.0/prose/csaf-v2-editor-draft.md

```diff
@@ -3352,6 +3350,125 @@ This does not exclude that such a document is also available in an access protec
 
 > Reasoning: If an advisory is already in the media, an end user should not be forced to collect the pieces of information from a press release but be able to retrieve the CSAF document.
 
+### 5.1.5 Requirement 5: security.txt
+
+In the security.txt there MUST be at least one field `CSAF` which points to either the ROLIE service document or a directory with CSAF files. If this field indicates a web URI, then it MUST begin with "https://" (as per section 2.7.2 of [RFC7230]).
+
+> At the time of this writing, the security.txt is still a proposed standard. The `CSAF` field has not been officially added yet.
+
+Examples:
+
+```
+CSAF: https://domain.tld/security/data/csaf/
+CSAF: https://psirt.domain.tld/advisories/csaf/
+CSAF: https://domain.tld/security/csaf/csaf-service.json
+```
+
+### 5.1.6 Requirement 6: DNS path
+
+The DNS record `csaf.data.security.domain.tld` SHALL resolve as a webserver which either serves directly the ROLIE service document or a directory with CSAF files.
+
+### 5.1.7 Requirement 7: One folder per year
+
+The CSAF documents must be located within folders named `<YYYY>` where `<YYYY>` is the year given in the value of `/document/tracking/initial_release_date`.
+
+Examples:
+
+```
+2021
+2020
+```
+
+### 5.1.8 Requirement 8: index.txt
+
+The index.txt file within MUST provide a list of all filenames of CSAF documents which are located in the sub-directories with their filenames.
+
+Examples:
+
+```
+2020/example_company_-_2020-yh4711.json
+2019/example_company_-_2019-yh3234.json
+2018/example_company_-_2018-yh2312.json
+```
+
+> This can be used to download all CSAF documents.
+
+### 5.1.9 Requirement 9: changes.csv
+
+The file changes.csv must contain the filename as well as the value of `/document/tracking/current_release_date` for each CSAF document in the sub-directories without a heading; lines must be sorted by the latter one.
+
+Examples:
+
+```
+2020/example_company_-_2020-yh4711.json, "2020-07-01T10:09:07Z"
+2018/example_company_-_2018-yh2312.json, "2020-07-01T10:09:01Z"
+2019/example_company_-_2019-yh3234.json, "2019-04-17T15:08:41Z"
+2018/example_company_-_2018-yh2312.json, "2019-03-01T06:01:00Z"
+```
+
+### 5.1.10 Requirement 10: Directory listings
+
+Directory listing SHALL be enabled to support manual navigation.
+
+### 5.1.11 Requirement 11: ROLIE service document
+
+The ROLIE service document MUST be a JSON file that conforms with [RFC8322].
+
+**TODO: Provide Example**
+
+### 5.1.12 Requirement 12: ROLIE category document
```

Comment:
I don't think I would make this a requirement because categories using within a Category document have to be registered:

https://datatracker.ietf.org/doc/html/rfc8322#section-8.4

For CSAF specifically, I could imagine registering "security advisory" as an information type:

https://www.iana.org/assignments/rolie/rolie.xhtml

and then a subregistry with all the CSAF document categories, but seeing as they are not enforced in the schema, the two sources of information (ROLIE registry vs CSAF schema) might get out of sync really fast.

All in all, you can publish ROLIE feeds without any categorization information.

Reply:
Ok, I think I read that wrong, the specific information types don't have to be registered as indicated by this example:

https://datatracker.ietf.org/doc/html/rfc8322#appendix-B.1

which uses the term "vulnerability" despite not being included in the list of potential types:

https://datatracker.ietf.org/doc/html/rfc8322#section-7.1.2

A global `urn:ietf:params:rolie:category:information-type` feed attribute actually determines whether feed is a ROLIE feed or not so the last sentence of my previous comment can be ignored. For CSAF feeds, it can I guess be set to:

```json
    "category": {
      "scheme": "urn:ietf:params:rolie:category:information-type",
      "term": "security advisory"
    }
```

In any case, category documents are not special in any way to ROLIE and are general to the Atom protocol specification:

https://datatracker.ietf.org/doc/html/rfc5023#section-7

We could consider coming up with guidance on how to publish category documents to cover some basic category types for 2.x, but I don't think I would include any category document requirements in 2.0.